src/firmware/main/world/firmware_robot.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "world/chicker.h"
+#include "world/dribbler.h"
+#include "world/wheel.h"
+
+/**
+ * This struct represents a robot from the perspective of firmware
+ */
+struct FirmwareRobot;
+typedef struct FirmwareRobot FirmwareRobot;
+
+/**
+ * Create a robot with the given hardware
+ *
+ * @param chicker The robot chicker
+ * @param dribbler The robot dribbler
+ * @param front_right_wheel The front right wheel of the robot
+ * @param front_left_wheel The front left wheel of the robot
+ * @param back_right_wheel The back right wheel of the robot
+ * @param back_left_wheel The back left wheel of the robot
```

Comment:
If the ball has XY coordinates, any reason why we don't give the robots its coordinates?

Reply:
That's a great point. I just took a look through the firmware to see what we're currently using and didn't see that the first time, but we must use it somewhere for control. Added.

Reply:
For reference, the robot's positional data is provided by `dr_get` in `dr.c`. Primitives call this function to get the current robot state.

Eg. `shoot.c`, line 145-146

Reply:
Interesting. I think the position belongs more in the Robot

Reply:
For sure in the new abstraction, totally agree